llvm/test/ThinLTO/X86/memprof-icp.ll

```diff
@@ -0,0 +1,320 @@
+;; Test that cloning of an indirect call works. We should perform ICP and update
+;; promoted call to the correct clone.
+
+;; This was created from the following source code, then the IR was reduced
+;; using llvm-reduce with the expected FileCheck input.
```

Comment:
I think we should also mention the step where the profile is collected and applied to the IR (which was then reduced)? I see the metadata has InstrProf so we should mention that here too.

@kazutakahirata I think here a text format profile would have helped significantly. The other profiles mixed in already have text formats so just because memprof doesn't it makes constructing the test even more tedious.